src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java

```diff
@@ -306,38 +323,64 @@ public String getSDF(BBChemParentStructure bbChemStructure) throws IOException{
-		// removeHs: (optional) toggles removing hydrogens from the molecule. This only make sense when sanitization is done. Defaults to true.
-		// strictParsing: (optional) if this is false, the parser is more lax about. correctness of the content. Defaults to true.
-		RWMol mol = RDKFuncs.MolBlockToMol(molstructure, false, false, false);
-
-		// Do partial sanization on mol so that we can calculate properties
-		// https://www.rdkit.org/docs/cppapi/namespaceRDKit_1_1MolOps.html#a1ea8c8a254b2f9cd006029e9ad72deac
-		// mol	: the RWMol to be cleaned
-		// operationThatFailed	: the first (if any) sanitization operation that fails is set here. The values are taken from the SanitizeFlags enum. On success, the value is SanitizeFlags::SANITIZE_NONE
-		// sanitizeOps	: the bits here are used to set which sanitization operations are carried out. The elements of the SanitizeFlags enum define the operations.
-		int ops = (SanitizeFlags.SANITIZE_ALL.swigValue() ^ SanitizeFlags.SANITIZE_CLEANUP.swigValue() ^
-		SanitizeFlags.SANITIZE_PROPERTIES.swigValue() ^ SanitizeFlags.SANITIZE_KEKULIZE.swigValue() ^
-		SanitizeFlags.SANITIZE_FINDRADICALS.swigValue() ^ SanitizeFlags.SANITIZE_CLEANUPCHIRALITY.swigValue());
-		RDKFuncs.sanitizeMol(mol, ops);
-
-		return mol;
-	}
+	public List<String> getMolFragments(String molfile) throws CmpdRegMolFormatException {
 
-	@Override
-	public String getMolStructureFromRDKMol(ROMol rdkMol) {
-		// https://www.rdkit.org/docs/cppapi/namespaceRDKit.html#a8159eb5c49b1d325ac994a15607cdffa
-		// MolToMolBlock
-		// mol	- the molecule in question
-		// includeStereo	- toggles inclusion of stereochemistry information
-		// confId	- selects the conformer to be used
-		// kekulize	- triggers kekulization of the molecule before it is written
-		// forceV3000	- force generation a V3000 mol block (happens automatically with more than 999 atoms or bonds)
-		return RDKFuncs.MolToMolBlock(rdkMol, true, -1, false, false);
+		// Read the preprocessor settings as json
+		JsonNode jsonNode = null;
+		try{
+			jsonNode = getPreprocessorSettings();
+		} catch (IOException e) {
+			logger.error("Error parsing preprocessor settings json");
+			throw new CmpdRegMolFormatException("Error parsing preprocessor settings json");
+		}
+
+		// Extract the processURL
+		JsonNode urlNode = jsonNode.get("splitURL");
+		if(urlNode == null || urlNode.isNull()) {
+			logger.error("Missing preprocessorSettings splitURL!!");
+		}
+		String url = urlNode.asText();
+
+		// Get the standardization actions and options
+		ObjectMapper mapper = new ObjectMapper();
+		ObjectNode requestData = mapper.createObjectNode();
+
+		// Add the structures to the request
+		ArrayNode arrayNode = mapper.createArrayNode();
+		arrayNode.add(molfile);
+		requestData.put("sdfs", arrayNode);
+
+		List<String> molStrings = new ArrayList<>();
+		// Post to the service and parse the response
+		try {
+			String requestString = requestData.toString();
```

Comment:
I actually don't think this will save much code at all.  A function call could replace these 2 lines in a few places, but I often use the `mapper` object to create new nodes and arrayNodes. 

```
		ObjectMapper mapper = new ObjectMapper();
		ObjectNode requestData = mapper.createObjectNode();
```